src/java/org/apache/cassandra/db/marshal/PartitionerDefinedOrder.java

```diff
@@ -84,7 +88,8 @@ public Term fromJSONObject(Object parsed)
     @Override
     public String toJSONString(ByteBuffer buffer, int protocolVersion)
     {
-        throw new UnsupportedOperationException();
+        assert baseType != null && !baseType.equals(this);
```

Comment:
This assertion fails if we try to read an sstable that has been written before applying this patch

Reply:
if reading from a sstable generated from the version that do not contains this patch , the baseType will be null and a java.lang.AssertionError will return at this line .

Reply:
Add some tests for this

Reply:
+1

Reply:
print serialized value seems not easy, for in PartitionerDefinedOrder class, lack of really data type. When  using fixed code to export older sstable.